src/components/mgt-person/mgt-person.ts

```diff
@@ -491,6 +497,8 @@ export class MgtPerson extends MgtTemplatedComponent {
     }
 
     if (this.personDetails) {
+      this._personAvatarBg = this.getColorFromName(this.personDetails.displayName);
```

Comment:
I notice that `this._personAvatarBg` is always set immediately after `this.personDetails` is set, in the `loadState()` function. Because of this, it might be better to do all of this work in the `renderImage()` function.

```
// add avatar background color
if (!this._personAvatarBg) {
  this._personAvatarBg = (this.personDetails.displayName)
    ? this.getColorFromName(this.personDetails.displayName)
    : 'gray20';
}
```

Another thought, we need to maintain the coupling of `this.personDetails` and `this._personAvatarBg` throughout the class; when `this.personDetails` changes, `this._personAvatarBg` must be updated as well. It might be easier to maintain this relationship by either:
1. Not using `this._personAvatarBg` and instead re-calculating whenever `this.renderImage` is called.
2. Updating `this.personAvatarBg` in the setter for `this.personDetails`.

Reply:
As background, I asked originally to remove the calculation from the render method because it only needs to be calculated when the `personDetails` change. The setter makes more sense in this case

Reply:
Sounds good!

Reply:
@shweaver-MSFT Ping... Updated 💃

Reply:
Love it!